File: src/php/core/src/Style/ComponentTranslate.php
<?php

declare(strict_types=1);

namespace DiceBear\Style;

/**
 * Read-only view over a component's `translate` block, providing the X and Y
 * offset ranges.
 */
class ComponentTranslate
{
    /**
     * @param array<string, mixed> $data
     */
    public function __construct(private readonly array $data) {}

    /** @return array{min: int|float, max: int|float, step?: int|float}|null */
    public function x(): ?array
    {
        return $this->data['x'] ?? null;
    }

    /** @return array{min: int|float, max: int|float, step?: int|float}|null */
    public function y(): ?array
    {
        return $this->data['y'] ?? null;
    }
}

File: src/rust/core/src/style/component_translate.rs
//! Read-only view over a component's `translate` block.

use serde::Deserialize;

use crate::prng::Range;

#[derive(Deserialize, Default)]
pub(crate) struct ComponentTranslate {
    x: Option<Range>,
    y: Option<Range>,
}

impl ComponentTranslate {
    pub(crate) fn x(&self) -> Option<Range> {
        self.x
    }

    pub(crate) fn y(&self) -> Option<Range> {
        self.y
    }
}
